### PR TITLE
Restrict React barrel imports

### PR DESCRIPTION
**Pull request by SebiVPS**
Use named imports instead of barrel importing React. Doing so will result in less code and better readability. The new [React docs](https://react.dev/) also use named imports and never import React itself.

---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: COM-1026

### CODE REVIEW

**Comment by johnnyomair** [comment]
@SebiVPS do you want me to review this? Normally I don't review draft PRs without explicit review request.

**Comment by SebiVPS** [comment]
> @SebiVPS do you want me to review this? Normally I don't review draft PRs without explicit review request.

@johnnyomair i would have pinged you, or set you as reviewer if i wanted too ;) 
I think we can wait for #2526 to be merged. (Would be nice if you could review that one ;) ) 
Thank you :)

**Comment by johnnyomair** [comment]
As discussed, we'll merge this just before releasing v8 (after next has been merged into main) to reduce the amount of conflicts.

File: storybook/.eslintrc.json
Comment (SebiVPS) [inline storybook/.eslintrc.json:13]: @johnnyomair because we still use react 17 for storybook, the react barrel import is still necessary i guess. I disabled no-unused-vars and no-unused-imports for the pipeline to run through. The alternative would be to add  (`// eslint-disable-next-line @typescript-eslint/no-unused-vars, unused-imports/no-unused-imports`) for every line, like i did for the handful imports in docs/ 

Is it planned to update storybook? Then the react imports should be removed and the rules can be deleted again.

Screenshot 2024-12-09 at 13 49 15

**Comment by SebiVPS** [comment]
> As discussed, we'll merge this just before releasing v8 (after next has been merged into main) to reduce the amount of conflicts.

@johnnyomair Manuel and Thomas said we can add this to next any time

File: storybook/.eslintrc.json
Comment (johnnyomair) [reply storybook/.eslintrc.json:13]: We've recently updated Storybook to 8.4. The new JSX Transform should already be supported since 6.1: https://storybook.js.org/blog/storybook-6-1/. So maybe we need to adapt the config somehow?

Comment (SebiVPS) [reply storybook/.eslintrc.json]: here we go, storybook updated

File: docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx
Comment (johnnyomair) [inline docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx:4]: Do we need the ESLint disables here? 🤔

**Review by johnnyomair — COMMENTED** [review]
We need an upgrade script for this.

File: docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx
Comment (SebiVPS) [reply docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx:4]: yes sadly ... the import can not be deleted with docs v2, need to update to docusaurus v3.

Comment (johnnyomair) [reply docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx:4]: But isn't the variable/import used then?

Comment (SebiVPS) [reply docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx:4]: 🤔  well, ofc its used, but it think because we extend the eslint rules from `"plugin:react/jsx-runtime"` its told that he should not need it.
Should we try to turn it off again for this package, or will it be updated anyway soon?

Comment (SebiVPS) [reply docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx:4]: I tried to adapt the eslint rules to disable the new JSX Transfrom again, but was not successful :( 
Can we leave it for these few lines?

**Comment by SebiVPS** [comment]
> We need an upgrade script for this.

We should try this: https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports

**Comment by SebiVPS** [comment]
> > We need an upgrade script for this.
> 
> We should try this: https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports

@johnnyomair i tried this codemod. I get best results, 

1. when replacing `import * as React from "react";` with `import React from "react";` 
2. then run the codemod: `npx react-codemod update-react-imports --force´ (force needed because of the changes before) 
3. then run lint with `--fix` `npm run lint:eslint --fix` 

it does not fix all the problems, but its much better than make all by hand.

**Comment by johnnyomair** [comment]
@SebiVPS could you please add a section to the [migration guide](https://github.com/vivid-planet/comet/blob/next/docs/docs/migration/migration-from-v7-to-v8.md) and add the steps there?

File: docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx
Comment (johnnyomair) [reply docs/src/stories/customization-and-styling/OverridingStylesOfAComponentGlobally.tsx:4]: Yes, let's leave it like this.

**Review by johnnyomair — COMMENTED** [review]
Please merge next and resolve conflicts, then this PR should be good to go.

File: packages/admin/admin-react-select/.eslintrc.json
Comment (johnnyomair) [inline packages/admin/admin-react-select/.eslintrc.json:1]: Please remove this file.

**Review by johnnyomair — CHANGES_REQUESTED** [review]

File: packages/admin/admin-react-select/.eslintrc.json
Comment (SebiVPS) [reply packages/admin/admin-react-select/.eslintrc.json:1]: oh, deleted

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**
